File: Editorial/1874_Minimize_Product_Sum_of_Two_Arrays/CPP/1_Sort_both_arrays.cpp
class Solution {
public:
    int minProductSum(vector<int>& nums1, vector<int>& nums2) {
        // Sort nums1 in ascending order, and nums2 in
        // descending order.
        sort(nums1.begin(), nums1.end());
        sort(nums2.begin(), nums2.end(), greater<int>());
        
        // Initialize sum as 0.
        int ans = 0;

        // Iterate over two sorted arrays and calculate the 
        // cumulative product sum.
        for (int i = 0; i < nums1.size(); ++i) {
            ans += nums1[i] * nums2[i];
        } 

        return ans;        
    }
};

File: Editorial/1801-1900/1874_Minimize_Product_Sum_of_Two_Arrays/Java/1_Sort_both_arrays.java
class Solution {
    public int minProductSum(int[] nums1, int[] nums2) {
        // Sort nums1 and nums2 in ascending order.
        Arrays.sort(nums1);
        Arrays.sort(nums2);

        // Initialize sum as 0.
        int ans = 0;
        int n = nums2.length;
        
        // Iterate over two sorted arrays and calculate the 
        // cumulative product sum. 
        for (int i = 0; i < n; ++i) {
            // Since we also sort nums2 in ascending order, 
            // we need to iterate over nums2 in reverse order.
            ans += nums1[i] * nums2[n - 1 - i];
        }

        return ans;
    }
}
